### PR TITLE
Make computer data correspond to individual tiles

### LINKED ISSUE #26481: "Workstation" from Refugee Center Quest Can Override All Other Computer Functions
**Describe the bug**  
When you get the second quest from the Refugee Center doctor, she will point you to a lab. Upon downloading the data, all of the computers in that lab that had not yet been interacted with will be converted into "Workstation" computers, regardless of their original intended function - making it impossible, for instance, to open metal doors by hacking. Completing the quest does not reset any already affected computers, but does seem to stop this from happening to new computers.

**To Reproduce**  
- Complete the refugee center quest for analyzing zombie blood and bringing it back. Turn in this quest and accept the next one.
- Go into a lab.
- Download the data from a computer designated as a workstation, but do not turn it in.
- Attempt to mess with any other computer that had not previously been touched. Look especially for those controlling doors.

**Expected behavior** 
Only the first computer the player tries to access with this quest active should be clobbered with the workstation functions, probably at runtime rather than overriding everything in the map (which may or may not be what it already does).

EDIT: I have realized that the way this quest seems to work is by spawning computers in the laboratory. In any case, that should be sufficient; replacing existing computers is superfluous.

**Versions and configuration(please complete the following information):**
 - OS: Windows 7
 - Game Version: 0.C-32651-gba0cd1a
 - Graphics version: Tiles
 - Mods loaded: Filthy Clothing, Simplified Nutrition, Craftable Gun Pack, Safe Autodoc, Bright Nights, Makeshift Items Mod, Mutant NPCs, NPC traits, More Locations, Tall Buildings, Boats, Folding Parts pack, Vehicle Additions Pack, Tanks and Other Vehicles, Classes and Scenarios Mod, No Acid Zombies, Fuji's More Buildings, Beta Natiional Guard Camp, Hair aka Rainbow, More Location hairs.

### PR DESCRIPTION
**Pull request by Qrox**
#### Summary
SUMMARY: Bugfixes "Fix existing consoles overwritten by mission consoles"

#### Purpose of change
This PR makes computer data correspond to individual tiles, so a submap can contain consoles of more than one function. Should fix #26481.

#### Describe the solution
`computer`s are now stored in a `std::map` member of `submap`, which maps submap locations to computer functions. Old saves are converted to the new system when computers are accessed or modified (except in the const `get_computer` function), and saved in the old or new format depending one whether this conversion has happened. This is because when loading the computers it's possible that the terrain (which is required for the conversion) has not been loaded yet. The code for placing mission consoles is also updated to better avoid creating consoles on existing consoles.

### CODE REVIEW

File: src/savegame_json.cpp [hunk 1]
```diff
@@ -3435,9 +3435,20 @@ void submap::store( JsonOut &jsout ) const
...
-        jsout.member( "computers", comp->save_data() );
+
+    if( legacy_computer ) {
+        // it's possible that no access to computers has been made and legacy_computer
+        // is not cleared
+        jsout.member( "computers", legacy_computer->save_data() );
+    } else if( !computers.empty() ) {
+        jsout.member( "computers" );
+        jsout.start_array();
+        for( auto &elem : computers ) {
```
Comment (nshcat) [inline src/savegame_json.cpp]: Doesnt this need a call to `start_object` and `end_object` for each loop iteration? You are constructing a array of objects here, correct?

File: src/savegame_json.cpp [hunk 2]
```diff
@@ -3435,9 +3435,20 @@ void submap::store( JsonOut &jsout ) const
...
+            jsout.write( elem.first.x );
+            jsout.write( elem.first.y );
```
Comment (BevapDin) [inline src/savegame_json.cpp]: Why write x and y manually? You can just serialize the `point` object directly.

Comment (Qrox) [reply src/savegame_json.cpp]: I didn't know that, it was done this way in the other functions, will look into it.

File: src/savegame_json.cpp (hunk 1 above)
Comment (Qrox) [reply src/savegame_json.cpp]: Not an array of objects, but an array of consecutive integers and strings, as it was done in nearby code.

**Pull request merged**